File: PyCheckio/split-pairs.py
def split_pairs(a):
    result = []
    st = ""
    count = 0
    if len(a) % 2 == 0:
        for i in range(0, len(a), 2):
            st = ""
            st += a[i] + a[i + 1]
            result.append(st)
        return result
    elif len(a) % 2 == 1:
        for i in range(len(a)):
            if count == 2:
                result.append(st)
                st = ""
                count = 0
            st += a[i]
            count += 1
        if count == 1:
            st += "_"
            result.append(st)
        return result
        


if __name__ == '__main__':
    print("Example:")
    print(list(split_pairs('abcd')))

    # These "asserts" are used for self-checking and not for an auto-testing
    assert list(split_pairs('abcd')) == ['ab', 'cd']
    assert list(split_pairs('abc')) == ['ab', 'c_']
    assert list(split_pairs('abcdf')) == ['ab', 'cd', 'f_']
    assert list(split_pairs('a')) == ['a_']
    assert list(split_pairs('')) == []
    print("Coding complete? Click 'Check' to earn cool rewards!")

File: JsCheckio/split-pairs.js
import assert from "assert";

function splitPairs(text: string): string[] {
    let test = []
    let part = ""
    if (text.length % 2 == 0) {
        for (let i = 0; i < text.length; i += 2) {
            part = text[i] + text[i + 1]
            test.push(part)
            }
        return test;
        }
    else {
        for (let i = 0; i < text.length - 1; i += 2) {
        part = text[i] + text[i + 1]
        test.push(part)
            }
            test.push(text[text.length - 1] + "_")
        return test;
        }
}

console.log('Example:');
console.log(splitPairs('abcd'));

// These "asserts" are used for self-checking
assert.deepEqual(splitPairs('abcd'), ['ab', 'cd']);
assert.deepEqual(splitPairs('abc'), ['ab', 'c_']);
assert.deepEqual(splitPairs('abcdf'), ['ab', 'cd', 'f_']);
assert.deepEqual(splitPairs('a'), ['a_']);
assert.deepEqual(splitPairs(''), []);

console.log("Coding complete? Click 'Check' to earn cool rewards!");
